### PR TITLE
Add random state to svd in _compute_pi function

### PR DESCRIPTION
**Pull request by agoscinski**
In one example we got nondeterministic behavior because of the random state in sparse linalg. We set the random state there now. The temporary fix to set np.random.seed is removed.

----
:books: Documentation preview :books:: https://scikit-matter--210.org.readthedocs.build/en/210/

### CODE REVIEW

File: src/skmatter/sample_selection/_base.py [hunk 1]
```diff
@@ -342,7 +342,6 @@ class CUR(_CUR):
     ...         [-0.03, -0.53, 0.08],
     ...     ]
     ... )
-    >>> np.random.seed(0)  # there is a source of randomness in it
     >>> selector.fit(X)
     CUR(n_to_select=2)
     >>> np.round(selector.pi_, 2)  # importance scole
```
Comment (PicoCentauri) [inline src/skmatter/sample_selection/_base.py:347]: On my machine I still get

```
Expected:
    array([0., 1., 0.])
Got:
    array([0.23, 0.  , 0.77])
```

for the output.

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
Thanks for the quick fix @agoscinski. I checked the code locally and now the output is constant but we still have to adjust the output.

File: src/skmatter/sample_selection/_base.py (hunk 1 above)
Comment (agoscinski) [reply src/skmatter/sample_selection/_base.py:347]: thanks for checking, I don't have a better idea than going back to debugging together comparing output of sparse svd on your and my machine

Comment (PicoCentauri) [reply src/skmatter/sample_selection/_base.py:347]: Okay then lets do it together next week.

**Review by PicoCentauri — APPROVED** [review]
Works now also on my machine!

**Pull request merged**